frontend/pages/admin/IntegrationsPage/cards/CertificateAuthorities/components/CustomESTForm/helpers.ts

```diff
@@ -0,0 +1,146 @@
+import { ICertificateAuthorityPartial } from "interfaces/certificates";
+
+import valid_url from "components/forms/validators/valid_url";
+
+import { ICustomESTFormData } from "./CustomESTForm";
+
+export interface ICustomESTFormValidation {
+  isValid: boolean;
+  name?: { isValid: boolean; message?: string };
+  url?: { isValid: boolean; message?: string };
+  username?: { isValid: boolean; message?: string };
+  password?: { isValid: boolean; message?: string };
+}
+
+type IMessageFunc = (formData: ICustomESTFormData) => string;
+type IValidationMessage = string | IMessageFunc;
+type IFormValidationKey = keyof Omit<ICustomESTFormValidation, "isValid">;
+
+interface IValidation {
+  name: string;
+  isValid: (formData: ICustomESTFormData) => boolean;
+  message?: IValidationMessage;
+}
+
+type IFormValidations = Record<
+  IFormValidationKey,
+  { validations: IValidation[] }
+>;
+
+export const generateFormValidations = (
+  customESTIntegrations: ICertificateAuthorityPartial[],
+  isEditing: boolean
+) => {
+  const FORM_VALIDATIONS: IFormValidations = {
+    name: {
+      validations: [
+        {
+          name: "required",
+          isValid: (formData: ICustomESTFormData) => {
+            return formData.name.length > 0;
+          },
+        },
+        {
+          name: "invalidCharacters",
+          isValid: (formData: ICustomESTFormData) => {
+            return /^[a-zA-Z0-9_]+$/.test(formData.name);
+          },
+          message:
+            "Invalid characters. Only letters, numbers and underscores allowed.",
+        },
+        {
+          name: "unique",
+          isValid: (formData: ICustomESTFormData) => {
+            return (
+              isEditing ||
+              customESTIntegrations.find(
+                (cert) => cert.name === formData.name
```

Comment:
Need to check the type here too -- we only validate uniqueness against certs of the same type